File: experiments/callgraph-precision/fixtures/extra_allocation/Entry.java
package probe;
public class Entry {
 public interface Step { void run(); }
 public static class Marker { public static boolean live; public static boolean dormant; }
 public static class Live implements Step { public void run(){ Marker.live=true; } }
 public static class Dormant implements Step { public void run(){ Marker.dormant=true; } }
 public static void execute(Step step){if(step!=null)step.run();}
 public static void main(String[] args) throws Exception { new Dormant(); Step step=new Live(); execute(step); }
}

File: experiments/callgraph-precision/fixtures/kotlin_extra/Entry.kt
package probe
object Entry {
 interface Step { fun run() }
 object Marker { @JvmField var live=false; @JvmField var dormant=false }
 class Live : Step { override fun run(){Marker.live=true} }
 class Dormant : Step { override fun run(){Marker.dormant=true} }
 @JvmStatic fun execute(step: Step?) { step?.run() }
 @JvmStatic fun main(args: Array<String>?) { Dormant(); val step: Step=Live(); execute(step) }
}
